File: go/2020/day1/main_test.go
package main

import "testing"

var (
	input = []int{
		1721,
		979,
		366,
		299,
		675,
		1456,
	}
)

func TestPart1(t *testing.T) {
	expected := 514579

	actual := part1(input)
	if actual != expected {
		t.Errorf("input [%v], actual [%v] != expected [%v]", input, actual, expected)
	}
}

func TestPart2(t *testing.T) {
	expected := 241861950

	actual := part2(input)
	if actual != expected {
		t.Errorf("input [%v], actual [%v] != expected [%v]", input, actual, expected)
	}
}

File: 2022/day8/js/main.test.js
"use strict";

const assert = require("assert");
const { part1, part2 } = require("./main");

const input = ["30373", "25512", "65332", "33549", "35390"];

async function test_part1(input) {
  const expected = 21;
  const actual = await part1(input);
  assert.equal(
    actual,
    expected,
    `actual: [${actual}] != expected [${expected}]`
  );
}

async function test_part2(input) {
  const expected = 8;
  const actual = await part2(input);
  assert.equal(
    actual,
    expected,
    `actual: [${actual}] != expected [${expected}]`
  );
}

async function test() {
  const data = input.map((r) => r.split(""));
  await test_part1(data);
  await test_part2(data);
}

test();
